packages/react-reconciler/src/ReactFiberCommitEffects.js

```diff
@@ -876,19 +932,21 @@ export function safelyDetachRef(
 function safelyCallDestroy(
   current: Fiber,
   nearestMountedAncestor: Fiber | null,
-  destroy: () => void,
+  destroy: mixed => void,
+  resource: mixed,
 ) {
+  const destroy_ = resource == null ? destroy : destroy.bind(null, resource);
```

Comment:
What if you return `null` as the resource?

Reply:
I have a dev check here: https://github.com/facebook/react/pull/31416/files/1a95853a7ccb6e04e5b4fdc077d708d42e66cb58#diff-6718a473fd7cd45dba73de7f6bef32a6bf0699b23d853d6698abf02cca59f56bR158-R164